include/gridtools/storage/adapter/python_sid_adapter.hpp

```diff
@@ -54,7 +54,25 @@ namespace gridtools {
             }
         };
 
-        template <class T, size_t Dim, class Kind, int UnitStrideDim>
+        template <size_t UnitStrideDim,
+            class Strides,
+            class Shape,
+            std::enable_if_t<(UnitStrideDim >= tuple_util::size<std::decay_t<Strides>>::value), int> = 0>
+        Strides &&assign_unit_stride(Strides &&strides, Shape &&) {
+            return std::forward<Strides>(strides);
+        }
+
+        template <size_t UnitStrideDim,
+            class Strides,
+            class Shape,
+            std::enable_if_t<(UnitStrideDim < tuple_util::size<std::decay_t<Strides>>::value), int> = 0>
+        decltype(auto) assign_unit_stride(Strides &&strides, Shape &&shape) {
+            return tuple_util::transform_index(
+                transform_strides_f<UnitStrideDim>{tuple_util::get<UnitStrideDim>(std::forward<Shape>(shape)) > 1},
```

Comment:
Let's add a comment here with link to `https://numpy.org/devdocs/release/1.8.0-notes.html#npy-relaxed-strides-checking`

Reply:
done